### PR TITLE
analyze movies

### LINKED ISSUE #250: [Feature Request] Add support for movies
Altough it is not (so easy) possible to automatically detect intros/outros on movies since you only have one media file you can't compare, it would still be nice to show them on the plugin config page. So you would be able to set those timestamps manually. Maybe the outro detection is possible using the blackframe analyzer..
As a consequence, the "Next episode" button at the end could then just close the player and be called "End movie" or so.

Maybe a bigger project for the future, not sure.
Thank you in advance!

### PR DESCRIPTION
**Pull request by rlauuzo**
Basic functionality for movie scanning. Builds on top of the existing functions used for Tv episodes. Is untested.

## Summary by Sourcery

Introduce movie analysis functionality, extending the existing TV episode analysis capabilities to include movies. Update the configuration interface to allow users to opt-in for movie analysis and adjust the backend to support movie processing.

New Features:
- Add functionality to analyze movies in addition to TV episodes, allowing users to include movies in the analysis process.

Enhancements:
- Update the configuration page to include an option for analyzing movies, enhancing user control over the analysis process.
- Modify the QueueManager to handle movie items, enabling the queuing and processing of movies for analysis.

### CODE REVIEW

File: ConfusedPolarBear.Plugin.IntroSkipper/Data/QueuedEpisode.cs [hunk 1]
```diff
@@ -47,6 +47,11 @@ public class QueuedEpisode
     /// </summary>
     public bool IsAnime { get; set; }
 
+    /// <summary>
+    /// Gets or sets a value indicating whether an episode is Anime.
```
Comment (AbandonedCart) [inline ConfusedPolarBear.Plugin.IntroSkipper/Data/QueuedEpisode.cs:51]: Friendly reminder

Comment (jumoog) [reply ConfusedPolarBear.Plugin.IntroSkipper/Data/QueuedEpisode.cs:51]: and the AI didn't catch it...

**Comment by rlauuzo** [comment]
autoskip works should work now.

**Comment by rlauuzo** [comment]
@sourcery-ai review

File: ConfusedPolarBear.Plugin.IntroSkipper/Manager/QueueManager.cs [hunk 2]
```diff
@@ -6,291 +6,304 @@
...
+                }
+            }
 
-        // Queue all episodes on the server for fingerprinting.
-        _logger.LogDebug("Iterating through library items");
+            _logger.LogDebug("Queued {Count} episodes", items.Count);
+        }
 
-        foreach (var item in items)
+        private void QueueMedia<T>(T media)
```
[the remark this answers is not in the document]
Comment (rlauuzo) [reply ConfusedPolarBear.Plugin.IntroSkipper/Manager/QueueManager.cs:162]: ?

**Comment by AbandonedCart** [comment]
Mostly only does credits, right? We’ll probably have to put a disclaimer down about it.

**Comment by jumoog** [comment]
nice 🥳

**Comment by rlauuzo** [comment]
**To-Do List:**
- Add setting: "Enable movie scanning?"
- Separate maximum credit length setting for movies
- Update metadata editor: Remove season and episode selection for movies
- Remove fingerprint visualization for movies

**Comment by AbandonedCart** [comment]
@sourcery-ai review

File: ConfusedPolarBear.Plugin.IntroSkipper/ScheduledTasks/BaseItemAnalyzerTask.cs [hunk 3]
```diff
@@ -212,7 +212,7 @@ private int AnalyzeItems(
             new ChapterAnalyzer(_loggerFactory.CreateLogger<ChapterAnalyzer>())
         };
 
-        if (first.IsAnime && Plugin.Instance!.Configuration.WithChromaprint)
+        if (first.IsAnime && Plugin.Instance!.Configuration.WithChromaprint && !first.IsMovie)
```
[the remark this answers is not in the document]
Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/ScheduledTasks/BaseItemAnalyzerTask.cs:215]: There is a reason

File: ConfusedPolarBear.Plugin.IntroSkipper/Manager/QueueManager.cs [hunk 4]
```diff
@@ -216,6 +225,47 @@ private void QueueEpisode(Episode episode)
             pluginInstance.TotalQueued++;
         }
 
+        private void QueueMovie(Movie movie)
```
[the remark this answers is not in the document]
Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/Manager/QueueManager.cs:228]: Comments are not user-facing.

File: ConfusedPolarBear.Plugin.IntroSkipper/Manager/QueueManager.cs [hunk 5]
```diff
@@ -141,13 +145,18 @@ private void QueueLibraryContents(Guid id)
...
             foreach (var item in items)
             {
-                if (item is not Episode episode)
+                if (item is Episode episode)
                 {
-                    _logger.LogDebug("Item {Name} is not an episode", item.Name);
-                    continue;
+                    QueueEpisode(episode);
+                }
+                else if (_analyzeMovies && item is Movie movie)
```
[the remark this answers is not in the document]
Comment (AbandonedCart) [reply ConfusedPolarBear.Plugin.IntroSkipper/Manager/QueueManager.cs:152]: This is a good suggestion overall, but overkill for the usage.

**Comment by Nekzuris** [comment]
> **To-Do List:**
> 
>     * [x]  Add setting: "Enable movie scanning"?

Where is this settings ?
Scanning my 1000+ movies on first install was a huge waste of time, I only want intro skipper for TV shows.

**Comment by AbandonedCart** [comment]
> Where is this settings ? Scanning my 1000+ movies on first install was a huge waste of time, I only want intro skipper for TV shows.

It's part of Jellyfin in 10.11 so you would have gone to the movies library in your dashboard, selected manage library, and then unchecked intro skipper under media segment providers.

**Comment by nyxaria** [comment]
@AbandonedCart this PR seems to be the only place this is documented

**Comment by AbandonedCart** [comment]
> @AbandonedCart this PR seems to be the only place this is documented

It's not a plugin feature. It should be in the jellyfin docs.

**Pull request merged**